src/java/org/apache/cassandra/service/CassandraDaemon.java

```diff
@@ -456,6 +476,78 @@ protected void setup()
         completeSetup();
     }
 
+    /**
+     * Checks if the data of the local system keyspaces need to be migrated to a different location.
+     *
+     * @throws IOException
+     */
+    private void migrateSystemDataIfNeeded() throws IOException
```

Comment:
It would be nice to have some upgrade dtests for this, provided that we have the machinery to do so in either Java or Python dtests.